app/helpers/hosts_and_hostgroups_helper.rb

```diff
@@ -87,10 +87,11 @@ def realm_field(f)
 
   def interesting_klasses(obj)
     classes    = obj.all_puppetclasses
-    smart_vars = LookupKey.reorder('').where(:puppetclass_id => classes.map(&:id)).group(:puppetclass_id).count
-    class_vars = LookupKey.reorder('').joins(:environment_classes).where(:environment_classes => { :puppetclass_id => classes.map(&:id) }).group('environment_classes.puppetclass_id').count
-    klasses    = smart_vars.keys + class_vars.keys
+    classes_ids = classes.reorder('').pluck(:id)
+    smart_vars = LookupKey.reorder('').where(:puppetclass_id => classes_ids).uniq.pluck(:puppetclass_id)
+    class_vars = LookupKey.reorder('').joins(:environment_classes).where(:environment_classes => { :puppetclass_id => classes_ids }).uniq.pluck('environment_classes.puppetclass_id')
+    klasses    = smart_vars + class_vars
```

Comment:
nice improvement! would you mind adding one more?
`klasses = (smart_vars + class_vars).uniq` otherwise we generate SQL like `AND "puppetclasses"."id" IN (60, 60)`